scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java

```diff
@@ -361,18 +362,20 @@ DTLSMessage getNextMessage() throws GeneralSecurityException, HandshakeException
 		 * @throws GeneralSecurityException if the record's ciphertext fragment could not be de-crypted 
 		 */
 		DTLSMessage getNextMessage(Record candidate) throws GeneralSecurityException, HandshakeException {
-			int epoch = candidate.getEpoch();
-			if (epoch < session.getReadEpoch()) {
-				// discard old message
-				LOGGER.debug("Discarding message from peer [{}] from past epoch [{}] < current epoch [{}]",
-						getPeerAddress(), epoch, session.getReadEpoch());
-				return null;
-			} else if (epoch == session.getReadEpoch()) {
-				candidate.applySession(session);
+			int recordEpoch = candidate.getEpoch();
+			int sessionEpoch = session.getReadEpoch();
+			if (recordEpoch == sessionEpoch) {
 				DTLSMessage fragment = candidate.getFragment();
+				if (fragment == null) {
+					LOGGER.debug("Cannot process message of type [{}], discarding...", candidate.getType());
+					return null;
+				}
+				if (fragment.getContentType() != ContentType.HANDSHAKE
+						&& fragment.getContentType() != ContentType.CHANGE_CIPHER_SPEC) {
+					LOGGER.debug("Drop message of type [{}], discarding...", candidate.getType());
```

Comment:
(same here)
I'm not sure but if this happened, this means there is a bug at scandium side ?

If yes, should we log ERROR or WARN ?